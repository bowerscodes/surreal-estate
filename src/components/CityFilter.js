import React from "react";
import { Link } from "react-router-dom";
import "../styles/cityfilter.css";

function CityFilter() {
  return (
    <div className="cityfilter">
      <Link to={`/?query={"city":"Manchester"}`} className="cityfilter-item">
        Manchester
      </Link>
      <Link to={`/?query={"city":"Liverpool"}`} className="cityfilter-item">
        Liverpool
      </Link>
      <Link to={`/?query={"city":"Leeds"}`} className="cityfilter-item">
        Leeds
      </Link>
      <Link to={`/?query={"city":"Sheffield"}`} className="cityfilter-item">
        Sheffield
      </Link>
    </div>
  );
}

export default CityFilter;
