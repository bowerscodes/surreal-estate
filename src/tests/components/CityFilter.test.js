import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import CityFilter from "../../components/CityFilter";

describe("CityFilter", () => {
  it("the CityFilter component renders correctly", () => {
    const { asFragment, getByText } = render(
      <Router>
        <CityFilter />
      </Router>
    );

    expect(asFragment).toMatchSnapshot();
    expect(getByText("Manchester")).toBeInTheDocument();
    expect(getByText("Liverpool")).toBeInTheDocument();
    expect(getByText("Leeds")).toBeInTheDocument();
    expect(getByText("Sheffield")).toBeInTheDocument();
  });
});
