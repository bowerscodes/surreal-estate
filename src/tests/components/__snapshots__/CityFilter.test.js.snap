// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityFilter the CityFilter component renders correctly 1`] = `[Function]`;
